### ISSUE TITLE
Refactor Panel styles to use subcomponent styles when Button is refactored and adds IButtonStyleProps

### ISSUE THREAD

**Issue by cliffkoh** [microsoft/fluentui#5689]
Refactor Dropdown styles to use subcomponent styles when Panel is converted to JS Styling. (At present it uses selectors and fragile global classnames).

**Comment by JasonGore** [comment]
Panel conversion issue is #5367 and its associated PR is #5823. However, implementation of subcomponent styles is blocked on Button conversion #5938, so I will marked this as blocked for now.

**Comment by Vitalius1** [comment]
This issue was tagged in a `todo` in `Panel` pointing that it waits for `Button` conversion when in fact initially it was about the `Dropdown`. I have a fix for the `Dropdown` in #7688 and so that I don't have to create a new issue and change the comments in `Panel` for it, I thought I can repurpose this one. Let me know if I still should close this one and create a new one. 

The comments in the `todo` are not exactly correct as it's not blocked by the conversion but by `Button` to support `IButtonStyleProps`, which I reflected in the new title of this issue. It might not be the case to support the `subComponentStyles` for the `Button` in here with the new Slots/Tokens pattern 🤔

**Comment by brandonthomas** [comment]
@Vitalius1 and @JasonGore we were able to do button styles as sub components by wrapping in a closure and then calling a function. Sort of a hacky way to do it but it looked like this:

`import { IDetailPanelNavStyles, IDetailPanelNavStyleProps } from './DetailPanelNav.types';
import { IButtonStyles, FontSizes } from 'office-ui-fabric-react';

export const getDetailPanelNavStyles = (props: IDetailPanelNavStyleProps): IDetailPanelNavStyles => {
  const { theme } = props;

  const baseButtonStyle: Partial<IButtonStyles> = {
    root: {
      padding: 0,
      height: 40,
      width: 40,
      selectors: {
        ':hover': {
          backgroundColor: theme.semanticColors.buttonBackgroundCheckedHovered
        },
        ':active': {
          backgroundColor: theme.palette.neutralTertiary
        }
      }
    },
    icon: {
      fontSize: FontSizes.small,
      margin: 0,
      color: theme.semanticColors.bodyText
    }
  };

  return {
    root: {
      display: 'flex',
      padding: 3,
      height: 40,
      position: 'relative'
    },
    backButtonWrapper: {
      flex: '1 1 auto'
    },
    subComponentStyles: {
      backButton: (() => {
        return baseButtonStyle;
      })(),
      closeButton: (() => {
        return baseButtonStyle;
      })(),
      refreshButton: (() => {
        return baseButtonStyle;
      })()
    }
  };
};`

types as follows:
`export interface IDetailPanelNavSubComponentStyles {
  /**
   * styles for the back button
   */
  backButton?: IStyleSet<IButtonStyles>;

  /**
   * styles for the refresh button
   */
  refreshButton?: IStyleSet<IButtonStyles>;

  /**
   * styles for the close button
   */
  closeButton?: IStyleSet<IButtonStyles>;
}`
